fedimint-client-rpc/src/lib.rs

```diff
@@ -0,0 +1,604 @@
+    async fn add_client(&self, client_name: String, client: ClientHandleArc) {
+        let mut clients = self.clients.lock().await;
+        clients.insert(client_name, client);
+    }
+
+    async fn get_client(&self, client_name: &str) -> Option<ClientHandleArc> {
+        let clients = self.clients.lock().await;
+        clients.get(client_name).cloned()
+    }
+
+    fn add_rpc_handle(&self, request_id: u64, handle: AbortHandle) {
+        let mut handles = self.rpc_handles.lock().unwrap();
+        if handles.insert(request_id, handle).is_some() {
+            tracing::error!("RPC CLIENT ERROR: request id reuse detected");
+        }
+    }
+
+    fn remove_rpc_handle(&self, request_id: u64) -> Option<AbortHandle> {
+        let mut handles = self.rpc_handles.lock().unwrap();
+        handles.remove(&request_id)
+    }
+
+    async fn client_builder(db: Database) -> Result<fedimint_client::ClientBuilder, anyhow::Error> {
+        let mut builder = fedimint_client::Client::builder(db).await?;
+        builder.with_module(MintClientInit);
+        builder.with_module(LightningClientInit::default());
+        builder.with_module(WalletClientInit(None));
+        builder.with_module(MetaClientInit);
+        builder.with_primary_module_kind(fedimint_mint_client::KIND);
+        Ok(builder)
+    }
+
+    /// Handle joining federation using unified database
+    async fn handle_join_federation(
+        &self,
+        invite_code: String,
+        client_name: String,
+    ) -> anyhow::Result<()> {
+        // Check if wallet mnemonic is set
+        let mnemonic = self.get_mnemonic_from_db().await?.ok_or_else(|| {
+            anyhow::anyhow!("No wallet mnemonic set. Please set or generate a mnemonic first.")
+        })?;
+
+        let unified_db = self.get_unified_database().await?;
+        let mut client_prefix = vec![DbKeyPrefix::ClientDatabase as u8];
+        client_prefix.extend_from_slice(client_name.as_bytes());
+        let client_db = unified_db.with_prefix(client_prefix);
+
+        let invite_code = InviteCode::from_str(&invite_code)?;
+        let federation_id = invite_code.federation_id();
+
+        // Derive federation-specific secret from wallet mnemonic
+        let federation_secret = self.derive_federation_secret(&mnemonic, &federation_id);
+
+        let builder = Self::client_builder(client_db).await?;
+        let client = Arc::new(
+            builder
+                .preview(&invite_code)
+                .await?
+                .join(RootSecret::StandardDoubleDerive(federation_secret))
+                .await?,
+        );
+
+        self.add_client(client_name, client).await;
+        Ok(())
+    }
+
+    async fn handle_open_client(&self, client_name: String) -> anyhow::Result<()> {
+        // Check if wallet mnemonic is set
+        let mnemonic = self.get_mnemonic_from_db().await?.ok_or_else(|| {
+            anyhow::anyhow!("No wallet mnemonic set. Please set or generate a mnemonic first.")
+        })?;
```

Comment:
```suggestion
        let mnemonic = self.get_mnemonic_from_db().await?.context("No wallet mnemonic set. Please set or generate a mnemonic first.")?;
```

and `use anyhow::Context as _;` in imports